src/app/util/attribute-storage.h

```diff
@@ -235,6 +235,18 @@ const EmberAfCluster * emberAfFindClusterInType(const EmberAfEndpointType * endp
 // Initial configuration
 void emberAfEndpointConfigure();
 
+// setup a dynamic endpoint's EmberAfEndpointType from a list of template clusters.
+//
+// This is a alternative to declaring dynamic endpoint metadata using DECLARE_DYNAMIC_* macros.
+//
+// As clusters to be used in dynamic endpoint setup need to be defined in ZAP anyway
+// (usually on a special endpoint which remains always disabled), the cluster's
+// metadata including all attributes already exists and can be re-used this way,
+// without error prone manual duplicating with DECLARE_DYNAMIC_*
+//
+CHIP_ERROR setupDynamicEndpointDeclaration(EmberAfEndpointType & endpointType, chip::EndpointId templateEndpointId,
```

Comment:
public methods should be `emberAf` prefixed for easy identification (since we are in C land rather than C++ namespaces)

Reply:
done